pipenv/patched/notpip/req/req_set.py

```diff
@@ -97,8 +97,11 @@ def make_abstract_dist(req_to_install):
     """
     if req_to_install.editable:
         return IsSDist(req_to_install)
-    elif req_to_install.link and req_to_install.link.is_wheel:
-        return IsWheel(req_to_install)
+    elif req_to_install.link:
+        if req_to_install.link.is_wheel:
+            return IsWheel(req_to_install)
+        elif req_to_install.link.ext == '.egg':
+            return IsEgg(req_to_install)
```

Comment:
I _think_ we can avoid the `IsEgg(DistAbstraction)` below if we just say:

```
elif req_to_install.link.ext == '.egg':
    return req_to_install.get_dist()
```
? or some variant on that

`InstallRequirement` objects can already do this for eggs, I think